src/System.Linq.Parallel/tests/PlinqModesTests.cs

```diff
@@ -11,157 +11,135 @@ namespace Test
-                easyQueries[i].Invoke(ParallelExecutionMode.Default, verifier);
-                if (!verifier.Passed)
+                foreach (var query in EasyUnorderedQueries())
                 {
-                    Assert.True(false, string.Format("Easy query {0} expected to run in parallel in default mode", i));
+                    foreach (ParallelExecutionMode mode in modes)
+                    {
+                        yield return new object[] { results[0], results[1], query, mode };
+                    }
                 }
             }
-
-            // Verify that all queries in 'easyQueries' always run in forced mode
-            for (int i = 0; i < easyQueries.Length; i++)
+            foreach (object[] results in Sources.Ranges(counts))
             {
-                Verifier verifier = new ParVerifier();
-                easyQueries[i].Invoke(ParallelExecutionMode.ForceParallelism, verifier);
-                if (!verifier.Passed)
+                foreach (var query in EasyOrderedQueries())
                 {
-                    Assert.True(false, string.Format("Easy query {0} expected to run in parallel in force-parallelism mode", i));
+                    foreach (ParallelExecutionMode mode in modes)
+                    {
+                        yield return new object[] { results[0], results[1], query, mode };
+                    }
                 }
             }
+        }
+
+        private static IEnumerable<Labeled<Action<ParVerifier, ParallelQuery<int>>>> HardQueries()
+        {
+            yield return Labeled.Label<Action<ParVerifier, ParallelQuery<int>>>("Select+Where+TakeWhile+ToArray", (verifier, query) => query.Select(x => verifier.Verify(x)).Where(x => true).TakeWhile((x, i) => true).ToArray());
+            yield return Labeled.Label<Action<ParVerifier, ParallelQuery<int>>>("Select+Where+TakeWhile+foreach", (verifier, query) => query.Select(x => verifier.Verify(x)).Where(x => true).TakeWhile((x, i) => true).ForEach());
+            yield return Labeled.Label<Action<ParVerifier, ParallelQuery<int>>>("Where+Select+ElementAt", (verifier, query) => query.Where(x => true).Select(x => verifier.Verify(x)).ElementAt(8));
+            yield return Labeled.Label<Action<ParVerifier, ParallelQuery<int>>>("Where+Select+foreach", (verifier, query) => query.Where(x => true).Select(x => verifier.Verify(x)).ForEach());
+        }
 
-            // Verify that all queries in 'easyQueries' always run in forced mode
-            for (int i = 0; i < hardQueries.Length; i++)
+        public static IEnumerable<object[]> HardQueryData(object[] counts, object[] modes)
+        {
+            // Test doesn't apply to DOP == 1.  It verifies that work is actually
+            // happening in parallel, which won't be the case with DOP == 1.
+            if (Environment.ProcessorCount == 1) yield break;
+
+            foreach (object[] results in UnorderedSources.Ranges(counts))
```

Comment:
I thought it had been failing for ordered queries (not being run in parallel even when forced), but now I'm not sure why.  It certainly isn't failing now.  Adding it.